Sources/MapboxMaps/Snapshot/Snapshotter.swift

```diff
@@ -105,52 +106,106 @@ public class Snapshotter {
                 return
             }
 
+            // Render attributions over the snapshot
+            let attributions = Attribution.parse(sourceAttributions)
+
+            let margin: CGFloat = 10
             let rect = CGRect(origin: .zero, size: uiImage.size)
-            let format = UIGraphicsImageRendererFormat()
-            format.scale = scale
-            let renderer = UIGraphicsImageRenderer(size: uiImage.size, format: format)
-            let compositeImage = renderer.image { rendererContext in
 
-                // First draw the snaphot image into the context
-                let context = rendererContext.cgContext
+            let (logoSize, text) = AttributionMeasure.logoAndAttributionThatFits(rect: rect,
+                                                                                 attributions: attributions,
+                                                                                 margin: margin)
 
-                if let cgImage = uiImage.cgImage {
-                    context.draw(cgImage, in: rect)
-                }
+            // Create views on the main thread
+            let logoView = LogoView(logoSize: logoSize)
 
-                let pointForCoordinate = { (coordinate: CLLocationCoordinate2D) -> CGPoint in
-                    let screenCoordinate = snapshot.screenCoordinate(for: coordinate)
-                    return CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
-                }
+            // Center logo horizontally if not enough space
+            let logoCenteredX = (rect.width - logoView.bounds.width)/2
```

Comment:
can't this be done with constraints. Where you align the logo center anchor with the rect center anchor. Would be more consistent than these calculations

Reply:
This is something that we should investigate in future: changing the snapshotter code to use views throughout. I agree it might make things simpler in some regards. There may be some secondary benefits for example: 
- being able to use visual effect views for blurring, rather than the manual method that is currently used
- reusing attribution for our main MapView

Reply:
So why not do it now?